locator/topology.cc

```diff
@@ -206,7 +206,7 @@ const node* topology::add_node(node_holder nptr) {
 }
 
 const node* topology::update_node(node* node, std::optional<host_id> opt_id, std::optional<inet_address> opt_ep, std::optional<endpoint_dc_rack> opt_dr, std::optional<node::state> opt_st, std::optional<shard_id> opt_shard_count) {
-    tlogger.debug("topology[{}]: update_node: {}: to: host_id={} endpoint={} dc={} rack={} state={}, at {}", fmt::ptr(this), node_printer(node),
+    tlogger.debug("topology[{}]: update_node: {}: to: host_id={} endpoint={} dc={} rack={} state={} shard_count={}, at {}", fmt::ptr(this), node_printer(node),
```

Comment:
to the posterities, the compile-time format string check still works here. the reason why the mismatch of the placeholder fields and the format argument was not identified is that, {fmt} does not try to check if all args are used, at compile-time it does format with the format string and its arguments, but the extraneous arguments does not break the formatting. it could error out on unused arguments, and it's be handy when we don't use [named arguments](https://fmt.dev/latest/api.html#named-arguments). but {fmt} does not differentiate the API using unnamed arguments from those using the named argument. otherwise it has to use a map tracking the refcount of the arguments in `parse_replacement_field()`, and error out if any of the arguments' refcount remains zero when `parse_format_string()` returns.